Errors look real:
```
======================================================================
ERROR: test_fixtures_by_id (casexml.apps.phone.tests.test_ota_fixtures.OtaFixtureTest)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/mnt/commcare-hq-ro/corehq/util/test_utils.py", line 232, in inner
    return helper(*args, **kwargs)
  File "/mnt/commcare-hq-ro/corehq/util/test_utils.py", line 205, in __call__
    call_with_settings(fn_with_pre_and_post, run_config.settings, args, kwargs)
  File "/mnt/commcare-hq-ro/corehq/util/test_utils.py", line 220, in call_with_settings
    fn(*args, **kwargs)
  File "/mnt/commcare-hq-ro/corehq/util/test_utils.py", line 201, in fn_with_pre_and_post
    self.fn(*args, **kwargs)
  File "corehq/ex-submodules/casexml/apps/phone/tests/test_ota_fixtures.py", line 113, in test_fixtures_by_id
    fixture_xml = generator.get_fixture_by_id('user-groups', self.restore_user, version=V2)
  File "corehq/ex-submodules/casexml/apps/phone/fixtures.py", line 82, in get_fixture_by_id
    fixtures = self._get_fixtures(None, fixture_id, user, version, last_sync, None)
  File "corehq/ex-submodules/casexml/apps/phone/fixtures.py", line 74, in _get_fixtures
    providers = self.get_providers(user, version, group, fixture_id)
  File "corehq/ex-submodules/casexml/apps/phone/fixtures.py", line 69, in get_providers
    providers = [provider for provider in providers if provider_matches(provider)]
  File "corehq/ex-submodules/casexml/apps/phone/fixtures.py", line 67, in provider_matches
    return provider.id == full_id or provider.id == prefix
AttributeError: 'LocationFixtureProvider' object has no attribute 'id'
```

--- a/corehq/apps/locations/fixtures.py
+++ b/corehq/apps/locations/fixtures.py
@@ -104,7 +104,7 @@ class HierarchicalLocationSerializer(object):
 
 
 class FlatLocationSerializer(object):
-    id = 'commtrack:locations_v2'
+    id = 'locations'
 
     def get_xml_nodes(self, restore_user, all_locations):
         if not toggles.FLAT_LOCATION_FIXTURE.enabled(restore_user.project.name):

--- a/corehq/apps/locations/fixtures.py
+++ b/corehq/apps/locations/fixtures.py
@@ -58,12 +58,10 @@ def should_sync_locations(last_sync, location_db, restore_user):
 
 
 class LocationFixtureProvider(object):
-    id = 'commtrack:locations'
 
-    def __init__(self, serializers=None):
-        if serializers is None:
-            serializers = [HierarchicalLocationSerializer(), FlatLocationSerializer()]
-        self.serializers = serializers
+    def __init__(self, id, serializer):
+        self.id = id
+        self.serializer = serializer
 
     def __call__(self, restore_user, version, last_sync=None, app=None):
         """
@@ -84,18 +82,13 @@ class LocationFixtureProvider(object):
         if not should_sync_locations(last_sync, all_locations, restore_user):
             return []
 
-        xml_nodes = []
-        for serializer in self.serializers:
-            xml_nodes.extend(serializer.get_xml_nodes(restore_user, all_locations))
-
-        return xml_nodes
+        return self.serializer.get_xml_nodes(self.id, restore_user, all_locations)
 
 
 class HierarchicalLocationSerializer(object):
-    id = 'commtrack:locations'
 
-    def get_xml_nodes(self, restore_user, all_locations):
-        root_node = Element('fixture', {'id': self.id, 'user_id': restore_user.user_id})
+    def get_xml_nodes(self, fixture_id, restore_user, all_locations):
+        root_node = Element('fixture', {'id': fixture_id, 'user_id': restore_user.user_id})
         root_locations = all_locations.root_locations
 
         if root_locations:
@@ -104,14 +97,12 @@ class HierarchicalLocationSerializer(object):
 
 
 class FlatLocationSerializer(object):
-    id = 'locations'
 
-    def get_xml_nodes(self, restore_user, all_locations):
+    def get_xml_nodes(self, fixture_id, restore_user, all_locations):
         if not toggles.FLAT_LOCATION_FIXTURE.enabled(restore_user.project.name):
             return []
 
-        root_node = Element('fixture', {'id': self.id, 'user_id': restore_user.user_id})
-
+        root_node = Element('fixture', {'id': fixture_id, 'user_id': restore_user.user_id})
         outer_node = Element('locations')
         root_node.append(outer_node)
         for location in sorted(all_locations.by_id.values(), key=lambda l: l.site_code):
@@ -131,7 +122,12 @@ class FlatLocationSerializer(object):
         return [root_node]
 
 
-location_fixture_generator = LocationFixtureProvider()
+location_fixture_generator = LocationFixtureProvider(
+    id='commtrack:locations', serializer=HierarchicalLocationSerializer()
+)
+flat_location_fixture_generator = LocationFixtureProvider(
+    id='locations', serializer=FlatLocationSerializer()
+)
 
 
 def _all_locations(user):

--- a/corehq/apps/locations/fixtures.py
+++ b/corehq/apps/locations/fixtures.py
@@ -58,6 +58,7 @@ def should_sync_locations(last_sync, location_db, restore_user):
 
 
 class LocationFixtureProvider(object):
+    id = 'commtrack:locations'
 
     def __init__(self, serializers=None):
         if serializers is None:

--- a/corehq/apps/locations/tests/data/expand_from_root_flat.xml
+++ b/corehq/apps/locations/tests/data/expand_from_root_flat.xml
@@ -1,4 +1,4 @@
-<fixture id="commtrack:locations_v2" user_id="{user_id}">
+<fixture id="locations" user_id="{user_id}">
     <locations>
         <location county_id="{suffolk_id}" id="{boston_id}" state_id="{massachusetts_id}" type="city">
             <name>Boston</name>

--- a/corehq/apps/locations/tests/test_location_fixtures.py
+++ b/corehq/apps/locations/tests/test_location_fixtures.py
@@ -25,7 +25,7 @@ from .util import (
     LocationTypeStructure,
 )
 from ..fixtures import _location_to_fixture, LocationSet, should_sync_locations, location_fixture_generator, \
-    LocationFixtureProvider, FlatLocationSerializer, HierarchicalLocationSerializer
+    flat_location_fixture_generator
 from ..models import SQLLocation, LocationType, Location
 
 DEPENDENT_APPS = [
@@ -71,8 +71,7 @@ class FixtureHasLocationsMixin(TestXmlMixin):
             for desired_location in desired_locations
         }  # eg: {"massachusetts_id" = self.locations["Massachusetts"].location_id}
 
-        serializer = FlatLocationSerializer() if flat else HierarchicalLocationSerializer()
-        generator = LocationFixtureProvider(serializers=[serializer])
+        generator = flat_location_fixture_generator if flat else location_fixture_generator
         fixture = ElementTree.tostring(generator(self.user, V2)[0])
         desired_fixture = self.get_xml(xml_name).format(
             user_id=self.user.user_id,

--- a/settings.py
+++ b/settings.py
@@ -544,6 +544,7 @@ FIXTURE_GENERATORS = {
     # fixtures that must be sent along with the phones cases
     'case': [
         "corehq.apps.locations.fixtures.location_fixture_generator",
+        "corehq.apps.locations.fixtures.flat_location_fixture_generator",
     ]
 }
 

